runtime-transaction/src/instruction_processor.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    crate::program_id_flags::ProgramIdFlag,
+    agave_transaction_view::static_account_keys_frame::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
+    core::borrow::Borrow, solana_pubkey::Pubkey, solana_sdk::transaction::TransactionError,
+    solana_svm_transaction::instruction::SVMInstruction,
+};
+
+pub trait InstructionProcessor<T>: Default {
+    type OUTPUT;
+
+    /// Process instructions individually given a `flag` that is derived from
+    /// the program id.
```

Comment:
Is Flag necessary? Can we use Visitor pattern that each "type" implements `process_instruction`?